core-bundle/contao/widgets/RowWizard.php

```diff
@@ -0,0 +1,278 @@
+				++$rowCount;
+			}
+		}
+
+		$header = $rows = array();
+
+		for ($i = 0, $c = \count($this->varValue); $i < $c; ++$i)
+		{
+			$columns = array();
+			$header = array();
+
+			foreach ($this->arrFields as $key => $options)
+			{
+				if (\is_array($options['input_field_callback'] ?? null))
+				{
+					$header[] = array();
+					$columns[] = System::importStatic($options['input_field_callback'][0])->{$options['input_field_callback'][1]}($this->objDca);
+					continue;
+				}
+
+				if (\is_callable($options['input_field_callback'] ?? null))
+				{
+					$header[] = array();
+					$columns[] = $options['input_field_callback']($this->objDca);
+					continue;
+				}
+
+				$widget = $this->prepareWidget($key, $this->varValue[$i][$key] ?? null, $options, $i);
+
+				if (null !== $widget)
+				{
+					if ('be_widget' === $widget->template)
+					{
+						$header[] = array('label' => $widget->label ?? '', 'mandatory' => $widget->mandatory);
+						$widget->label = null;
+						$widget->template = $this->strTemplate;
+					}
+					else
+					{
+						$header[] = array();
+					}
+
+					$columns[] = $widget->generateWithError(true);
+				}
+			}
+
+			$rows[$i] = array(
+				'columns' => $columns,
+				'controls' => array(
+					'enable' => $this->varValue[$i]['enable'] ?? false,
+					'edit' => ($this->varValue[$i]['id'] ?? 0) > 0,
+				),
+			);
+		}
+
+		return System::getContainer()->get('twig')->render('@Contao/backend/widget/row_wizard.html.twig', array(
+			'id' => $this->strId,
+			'style' => $this->style,
+			'header' => $header,
+			'rows' => $rows,
+			'min_rows' => $this->min,
+			'max_rows' => $this->max,
+			'sortable' => $this->sortable,
+			'actions' => $this->actions,
+		));
+	}
+
+	public static function getAttributesFromDca($arrData, $strName, $varValue = null, $strField = '', $strTable = '', $objDca = null): array
+	{
+		$attributes = parent::getAttributesFromDca($arrData, $strName, $varValue, $strField, $strTable, $objDca);
+
+		if (isset($arrData['fields']) && !isset($attributes['fields']))
+		{
+			$attributes['fields'] = $arrData['fields'];
+		}
+
+		return $attributes;
+	}
+
+	private function prepareWidget(string $type, mixed $value, array $options, int $increment): Widget|null
+	{
+		if (isset($this->widgets[$increment][$type]))
+		{
+			return $this->widgets[$increment][$type];
+		}
+
+		if (!isset($options['inputType']))
+		{
+			return null;
+		}
+
+		/** @var class-string<Widget> $widgetClass */
+		$widgetClass = $GLOBALS['BE_FFL'][$options['inputType']];
+
+		if (!class_exists($widgetClass))
+		{
+			return null;
+		}
+
+		$data = $widgetClass::getAttributesFromDca($options, $type, $value, $this->strField, $this->strTable, $this->objDca);
+
+		$data['name'] = $this->strId . '[' . $increment . '][' . $data['name'] . ']';
+
+		if (\in_array($data['type'] ?? null, array('checkbox', 'label'), true))
+		{
+			$data['id'] = $data['name'];
+		}
+		else
+		{
+			$data['id'] .= '_' . $increment;
+		}
```

Comment:
```suggestion
		$name = $this->strId . '[' . $increment . '][' . $type . ']';
		$field = $this->strField . '[' . $increment . '][' . $type . ']';
		$data = $widgetClass::getAttributesFromDca($options, $name, $value, $field, $this->strTable, $this->objDca);
```

I think it might be more future proof to pass the correct name and field to the widgets. This way a Widget that needs specific handling gets passed the information required to do so.

Reply:
This change would not render the label for checkboxes and inputradio inline anymore, converting them back to a column header and making them less accessible.